### DISCUSSION TITLE
Asterisk character `*` not working in utility name

### DISCUSSION THREAD

**Discussion by brandonmcconnell** [tailwindlabs/tailwindcss#14238] [Help] [answered]
<details><summary> Technical Configuration</summary>
<br>
<table>
    <tr></tr>
    <tr>
        <th align="left">What version of Tailwind CSS are you using?</th>
    </tr>
    <tr>
        <td>v3.4.10</td>
    </tr>
    <tr>
        <th align="left">What build tool (or framework if it abstracts the build tool) are you using?</th>
    </tr>
    <tr>
        <td>Tailwind Play</td>
    </tr>
    <tr>
        <th align="left">What version of Node.js are you using?</th>
    </tr>
    <tr>
        <td>N/A</td>
    </tr>
    <tr>
        <th align="left">What browser are you using?</th>
    </tr>
    <tr>
        <td>Chrome v127.0.6533.120 (Official Build) (arm64)</td>
    </tr>
    <tr>
        <th align="left">What operating system are you using?</th>
    </tr>
    <tr>
        <td>macOS Sonoma v14.5 (23F79)</td>
    </tr>
</table>

</details>

---

**Reproduction URL**

https://play.tailwindcss.com/1hJWSCKURf?file=config

**Describe your issue**

I am making some improvements to my [tailwindcss-directional-shadows](https://github.com/brandonmcconnell/tailwindcss-directional-shadows) plugin, and a new direction I am adding is `*` that will apply the shadow to all directions.

However, any way I tried to configure the utilities to support any with the name `shadow-*-SIZE`, it didn't work. Changing `*` to `all` resolves the issue, but I would prefer to use `*` here, and I am not sure why it doesn't work.

<img src="https://github.com/user-attachments/assets/90e05520-cd8b-4cd2-ad97-5e856e02baf5" width="500" alt="" />

**Comment by wongjn** [comment, marked as answer]
You'd want to escape the asterisk character, since it has special meaning in a CSS selector: https://play.tailwindcss.com/daREawczFw?file=config.

**Reply by brandonmcconnell** [reply]
Thanks, great solution! I forgot this was a factor 😄
